### ISSUE TITLE
nativefier not able to create any app with its icon

### ISSUE THREAD

**Issue by champagnealexandre** [nativefier/nativefier#1277]
**Homework**

- [X] I looked at `nativefier --help` and https://github.com/nativefier/nativefier/blob/master/API.md
- [X] I searched existing issues, open & closed. Yes, my question is new.
- [X] I'm using the latest version available at https://github.com/nativefier/nativefier/releases

**Your question**

Somehow, my installation of nativefier isn’t able to create any app with the website’s icon.

For example...

**Steps to reproduce**

Running ```nativefier "www.google.com" --verbose``` leads to:

```
...
Converting icons...
Converting icon /var/folders/_d/msvqcfz913l3gglp7npjns8r0000gn/T/nativefier-23-25-23-iconinfer--58726-wClBdcu2TQiv/icon.png to /var/folders/_d/msvqcfz913l3gglp7npjns8r0000gn/T/nativefier-23-25-23-iconconv--58726-X9nL7CUOcuom/icon.icns. Calling shell command: "/usr/local/Cellar/nativefier/45.0.0/libexec/lib/node_modules/nativefier/icon-scripts/convertToIcns" "/var/folders/_d/msvqcfz913l3gglp7npjns8r0000gn/T/nativefier-23-25-23-iconinfer--58726-wClBdcu2TQiv/icon.png" "/var/folders/_d/msvqcfz913l3gglp7npjns8r0000gn/T/nativefier-23-25-23-iconconv--58726-X9nL7CUOcuom/icon.icns"
Failed to convert icon to .icns, skipping. Error: Icon conversion failed with status code 1.
stdout:
stderr: mktemp: too few X's in template ‘nativefier_icons’

    at iconShellHelper (/usr/local/Cellar/nativefier/45.0.0/libexec/lib/node_modules/nativefier/src/helpers/iconShellHelpers.ts:46:11)
    at Object.convertToIcns (/usr/local/Cellar/nativefier/45.0.0/libexec/lib/node_modules/nativefier/src/helpers/iconShellHelpers.ts:83:10)
    at Object.convertIconIfNecessary (/usr/local/Cellar/nativefier/45.0.0/libexec/lib/node_modules/nativefier/src/build/buildIcon.ts:87:24)
    at Object.buildNativefierApp (/usr/local/Cellar/nativefier/45.0.0/libexec/lib/node_modules/nativefier/src/build/buildNativefierApp.ts:176:3)
Copying icons if necessary
No icon specified in options; aborting
...
```

**Debug info**

I’ve tried both installing nativefier through npm and brew, same result.

I think the error is related to this line in ```convertToIcns```

```
TEMP_DIR="$(mktemp -d -t nativefier_icons)"
```

i.e. it doesn’t seem to comply with the ```mktemp``` syntax, at least on my system — but nothing seems out of the ordinary with my version of ```mktemp```:

```
alexandre@Alexandres-MacBook-Pro Desktop mktemp --version
mktemp (GNU coreutils) 8.32
```

Finally, been searching for a moment about this and I suppose this bug is specific to my machine — hence the reason I did not fill this as a bug report but as a question. In any case, it seems to exceed my knowledge of OSX !

**Context**

 - Nativefier: 45.0.0
 - Node.js: v14.17.3
 - Npm: 6.14.13
 - OS: Big Sur 11.5.2 (20G95)
 - Additional context: none

**Comment by champagnealexandre** [comment]
Googling a bit about this ```mktemp``` that I never had to use before led me to replace the line in ```convertToIcns``` that reads
```
TEMP_DIR="$(mktemp -d -t nativefier_icons)"
```
by
```
TEMP_DIR="$(mktemp -d -t nativefier_iconsXXXXXX)"
```
and suddenly all my apps have icons!

Anyone knows why I had to replace that line?

**Comment by jose** [comment]
@champagnealexandre, what do you mean by "replace the line in `convertToIcns`"?  Where can I find `convertToIcns`?

**Comment by jose** [comment]
oh, [this line](https://github.com/nativefier/nativefier/blob/fa9bd2aba56bad30f06a11919c752ea730961874/icon-scripts/convertToIcns#L41), right?

**Comment by champagnealexandre** [comment]
yes exactly!

you’ll have to do this again if you update nativefier, don’t know why this is happening

**Comment by ronjouch** [comment]
@champagnealexandre this one may be an easy fix, looks like a Big Sur coreutils annoyance.

Can you share here your contents of `mktemp --help` and `man mktemp` ?

**Comment by jose** [comment]
> you’ll have to do this again if you update nativefier, don’t know why this is happening

that requires one to compile/install nativefier I would like to avoid that and install it with brew.

> @champagnealexandre this one may be an easy fix, looks like a Big Sur coreutils annoyance.
> Can you share here your contents of mktemp --help and man mktemp ?

It might be, yes.  When I run `mktemp` on the command line I'm actually running the one from core utils (i.e., `/usr/local/opt/coreutils/libexec/gnubin/mktemp`)

```
$ which mktemp
/usr/local/opt/coreutils/libexec/gnubin/mktemp

$ mktemp --help
Usage: mktemp [OPTION]... [TEMPLATE]
Create a temporary file or directory, safely, and print its name.
TEMPLATE must contain at least 3 consecutive 'X's in last component.
If TEMPLATE is not specified, use tmp.XXXXXXXXXX, and --tmpdir is implied.
Files are created u+rw, and directories u+rwx, minus umask restrictions.

  -d, --directory     create a directory, not a file
  -u, --dry-run       do not create anything; merely print a name (unsafe)
  -q, --quiet         suppress diagnostics about file/dir-creation failure
      --suffix=SUFF   append SUFF to TEMPLATE; SUFF must not contain a slash.
                        This option is implied if TEMPLATE does not end in X
  -p DIR, --tmpdir[=DIR]  interpret TEMPLATE relative to DIR; if DIR is not
                        specified, use $TMPDIR if set, else /tmp.  With
                        this option, TEMPLATE must not be an absolute name;
                        unlike with -t, TEMPLATE may contain slashes, but
                        mktemp creates only the final component
  -t                  interpret TEMPLATE as a single file name component,
                        relative to a directory: $TMPDIR, if set; else the
                        directory specified via -p; else /tmp [deprecated]
      --help     display this help and exit
      --version  output version information and exit

GNU coreutils online help: <https://www.gnu.org/software/coreutils/>
Full documentation <https://www.gnu.org/software/coreutils/mktemp>
or available locally via: info '(coreutils) mktemp invocation'
```

**Comment by ronjouch** [comment]
> It might be, yes. When I run `mktemp` on the command line I'm actually running the one from core utils (i.e., `/usr/local/opt/coreutils/libexec/gnubin/mktemp`)

@champagnealexandre 👍.

I don't have a Mac / Big Sur here. Can you also paste the `mktemp --help` of your stock OS / non-homebrew-coreutils one? (I mean the one not located in /usr/local/opt`, the one probably somewhere in `/usr/bin` or `/usr/local/bin`

**Comment by jose** [comment]
```
$ /usr/bin/mktemp --help
/usr/bin/mktemp: illegal option -- -
usage: mktemp [-d] [-q] [-t prefix] [-u] template ...
       mktemp [-d] [-q] [-u] -t prefix
```

**Comment by ronjouch** [comment]
Should be fixed, please test Nativefier 45.0.6, on npm in a few minutes/hours.
